# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/workers/WorkerEventTarget.h
# Commit: 07833c96f60c
# Full Hash: 07833c96f60cba643d641aaa04709574e7d69bb9
# Author: Eden Chuang <echuang@mozilla.com>
# Date: 2025-03-26 09:13:23
# Regressor Bug: 1899503
# File Overlap Count: 4
# Description:
#   Bug 1899503 - P3 PRemoteWorkerDebugger binding implementation. r=asuth
#   
#   This patch implements the PRemoteWorkerDebugger binding.
#   
#   We are trying to keep the same logic debugger registration/unregistration to minimize the difference between the local debugger(the debugger on the content process main thread)
# ==============================================================================

diff -r df07abc82381 -r 07833c96f60c dom/workers/WorkerEventTarget.h
--- a/dom/workers/WorkerEventTarget.h	Tue Mar 25 14:33:27 2025 +0000
+++ b/dom/workers/WorkerEventTarget.h	Tue Mar 25 14:33:28 2025 +0000
@@ -24,7 +24,9 @@
   //   with a holder in place.)
   //
   // * ControlOnly targets will simply dispatch a control runnable.
-  enum class Behavior : uint8_t { Hybrid, ControlOnly };
+  //
+  // * DebuggerOnly targets will simply dispatch a debugger runnable.
+  enum class Behavior : uint8_t { Hybrid, ControlOnly, DebuggerOnly };
 
  private:
   mozilla::Mutex mMutex;